File: src/0345.reverse-vowels-of-a-string.345/solution.ts
function reverseVowels (s: string): string {
    const list = s.split('');
    let left = 0;
    let right = s.length - 1;
    while (left < right) {
        while (left < right && !isVowel(list[left])) {
            left++;
        }
        while (left < right && !isVowel(list[right])) {
            right--;
        }
        if (left < right) {
            const tmp = list[left];
            list[left++] = list[right];
            list[right--] = tmp;
        }
    }
    return list.join('');
}
const set = new Set<string>([
    'a',
    'e',
    'i',
    'o',
    'u',
    'A',
    'E',
    'I',
    'O',
    'U',
]);

function isVowel (c:string) {
    return set.has(c);
}

File: src/0345.reverse-vowels-of-a-string.345/solution.java
class Solution {
    public String reverseVowels(String s) {
        char[] chars = s.toCharArray();
        int l = 0;
        int r = chars.length - 1;
        while (l < r) {
            while (l < r && !isVowel(chars[l])) {
                l++;
            }

            while (l < r && !isVowel(chars[r])) {
                r--;
            }

            if (l < r) {
                char tmp = chars[l];
                chars[l++] = chars[r];
                chars[r--] = tmp;
            }
        }

        return String.valueOf(chars);
    }

    public boolean isVowel(char c) {
        return "aeiouAEIOU".indexOf(c) > -1;
    }
}